File: Leetcode/2900.py
from typing import List

class Solution:
    def getLongestSubsequence(self, words: List[str], groups: List[int]) -> List[str]:
        n=len(words)
        last=groups[0]
        indices:List[int]=[0]

        for idx in range(1,n):
            if groups[idx]!=groups[indices[-1]]:
                indices.append(idx)
        return [words[i] for i in indices]

File: Leetcode/2900.go
package Leetcode

func getLongestSubsequence(words []string, groups []int) []string {
	n := len(words)
	ans := make([]string, 0, n)
	last := -1
	for idx, word := range words {
		if groups[idx] != last {
			last = groups[idx]
			ans = append(ans, word)
		}
	}
	return ans
}
